tests/test_iterative.py

```diff
@@ -0,0 +1,78 @@
+import io
```

Comment:
Rename file, I assume "iterative" stands for the iterative nature of the basic parser but it's really confusing.